e2e/nomostest/gitproviders/util/reponame.go

```diff
@@ -22,19 +22,41 @@ import (
 )
 
 const (
-	repoNameMaxLen  = 63
-	repoNameHashLen = 8
+	defaultRepoNameMaxLen   = 63
+	bitbucketRepoNameMaxLen = 62
+	repoNameHashLen         = 8
 )
 
-// SanitizeRepoName replaces all slashes with hyphens, and truncate the name.
+// SanitizeGCPRepoName replaces all slashes with hyphens, and truncates the name.
 // repo name may contain between 3 and 63 lowercase letters, digits and hyphens.
-func SanitizeRepoName(repoPrefix, name string) string {
+// This is used by CSR and SSM which are per-project resources
+func SanitizeGCPRepoName(repoPrefix, name string) string {
 	fullName := "cs-e2e-" + repoPrefix + "-" + name
+	hashStr := hashName(fullName)
```

Comment:
Minor nitpick: what happens if name or repoSuffix is an empty string? Same question applies to both repo